src/track/beatfactory.cpp

```diff
@@ -1,10 +1,68 @@
+#include <QFile>
 #include <QtDebug>
 #include <QStringList>
 
 #include "track/beatgrid.h"
 #include "track/beatmap.h"
 #include "track/beatfactory.h"
 #include "track/beatutils.h"
+#include "util/cmdlineargs.h"
+
+
+namespace {
+
+void debugBeats(const Track& track, const QVector<double>& rawBeats,
+        const QVector<double>& correctedBeats, QString beatsVersion, QString beatsSubVersion) {
+    if(!CmdlineArgs::Instance().getAnalyzerDebug()) {
+        return;
+    }
+    QString debugFilename = QDir(CmdlineArgs::Instance().getSettingsPath()).filePath("beatAnalyzerOutput.csv");
+    QFile debugFile(debugFilename);
+    if (!debugFile.open(QIODevice::Append | QIODevice::Text)) {
+        qWarning() << "ERROR: Could not open debug file:" << debugFilename;
+        return;
+    }
+    QString trackHeader;
+    trackHeader += track.getInfo();
+    trackHeader += ", analysed at ";
+    trackHeader += QDateTime::currentDateTime().toString("yyyy-MM-dd_hh'h'mm'm'ss's'");
+    trackHeader += ", with " + beatsVersion + beatsSubVersion;
+    debugFile.write(trackHeader.toLocal8Bit());
+    QString sRawBeats;
+    QString sRawBeatLenght;
+    auto previousBeat = rawBeats.begin();
+    sRawBeats += QString::number(*previousBeat, 'f') + ",";
+    for (auto beat = std::begin(rawBeats) + 1, end = std::end(rawBeats); beat != end; beat += 1) {
+        sRawBeats += QString::number(*beat, 'f') + ",";
+        sRawBeatLenght += QString::number(*beat - *previousBeat, 'f') + ",";
+        previousBeat = beat; 
+    }
+    QString sCorrectedBeats;
+    QString sCorrectedBeatLenght;
+    previousBeat = correctedBeats.begin();
+    sCorrectedBeats += QString::number(*previousBeat, 'f') + ",";
+    for (auto beat = std::begin(correctedBeats) + 1, end = std::end(correctedBeats); beat != end; beat += 1) {
+        sCorrectedBeats += QString::number(*beat, 'f') + ",";
+        sCorrectedBeatLenght += QString::number(*beat - *previousBeat, 'f') + ",";
+        previousBeat = beat; 
+    }
+    QString resultHeader = "\nRaw beats\n";
+    debugFile.write(resultHeader.toLocal8Bit());
+    debugFile.write(sRawBeats.toLocal8Bit());
+    resultHeader = "\nCorrected beats\n";
+    debugFile.write(resultHeader.toLocal8Bit());
+    debugFile.write(sCorrectedBeats.toLocal8Bit());
+    resultHeader = "\nRaw beat lenght\n";
+    debugFile.write(resultHeader.toLocal8Bit());
+    debugFile.write(sRawBeatLenght.toLocal8Bit());
+    resultHeader = "\nCorrected beat lenght\n";
+    debugFile.write(resultHeader.toLocal8Bit());
+    debugFile.write(sCorrectedBeatLenght.toLocal8Bit());
+    debugFile.write("\n");
+    debugFile.close();
+}
```

Comment:
Is this code supposed to be merged?

Reply:
I would say so, it`s protected under a new command-line argument "--analyzerDebug" and will be going very useful for future improvements on this...

Reply:
We should use Qt logging categories instead of command line options... #2782

Reply:
I'm fine with using the command line option for now though.